teuthology/nuke/__init__.py

```diff
@@ -296,7 +296,7 @@ def nuke_one(ctx, target, should_unlock, synch_clocks, reboot_all,
         ret = target
     else:
         if should_unlock:
-            unlock_one(ctx, target.keys()[0], ctx.owner)
+            unlock_one(ctx, list(target.keys())[0], ctx.owner)
```

Comment:
not sure how many elements `target.keys()` has, but if there are a lot, it's faster to really just get the first element with `next(iter(target.keys()))` instead of keeping all keys in memory just to get the first one.
Same comment is true for other places where you do that.
But I'm fine with keeping it as it is if the list is not super large.

Reply:
You're right, and sometime I use next-iter combination, I think it is not critical here since keys are just an objects.